pkg/server/handler/recordutil.go

```diff
@@ -26,6 +27,45 @@ func injectSigner(record *skydb.Record, store asset.Store) {
 	}
 }
 
+// scrubRecordFieldsForRead checks the field ACL to remove the fields
+// from a skydb.Record that the user is not allowed to read.
+func scrubRecordFieldsForRead(userInfo *skydb.UserInfo, record *skydb.Record, fieldACL skydb.FieldACL) {
+	for _, key := range record.UserKeys() {
+		if !fieldACL.Accessible(record.ID.Type, key, skydb.ReadFieldAccessMode, userInfo, record) {
+			record.Remove(key)
+		}
+	}
+}
+
+// scrubRecordFieldsForWrite checks the field ACL for write access.
+// Depending on whether the request is an atomic one, this function
+// will either remove the fields if the user is not allowed access if atomic
+// is false, or will return an error.
+func scrubRecordFieldsForWrite(userInfo *skydb.UserInfo, record *skydb.Record, origRecord *skydb.Record, fieldACL skydb.FieldACL, atomic bool) skyerr.Error {
+	nonWritableFields := []string{}
+
+	var deltaRecord skydb.Record
+	deriveDeltaRecord(&deltaRecord, origRecord, record)
+
+	for key := range deltaRecord.Data {
+		if fieldACL.Accessible(record.ID.Type, key, skydb.WriteFieldAccessMode, userInfo, origRecord) {
+			continue
+		}
+
+		if atomic {
+			nonWritableFields = append(nonWritableFields, key)
+			continue
+		}
+
+		record.Remove(key)
+	}
+
+	if len(nonWritableFields) > 0 {
+		return skyerr.NewDeniedArgument("Unable to save to some record fields because of Field ACL denied update.", nonWritableFields)
+	}
+	return nil
+}
```

Comment:
I hope I am not mistaken, you are talking about `scrubRecordFieldsForRead ` and `scrubRecordFieldsForWrite`, right?

I am not sure how to put the logic under `skydb.Record` because that will make the `skydb.Record` dependent on `skydb.FieldACL` and `skydb.UserInfo`. `skydb.Record` has been simple and it is not dependent on other struct.